packages/application-shell-connectors/src/components/application-context/application-context.spec.tsx

```diff
@@ -1,141 +1,172 @@
-            env: 'development',
-            cdnUrl: 'http://localhost:3001',
-            servedByProxy: false,
-            // extra props
-            foo: 'bar',
-          }}
-        >
-          <div />
-        </ApplicationContextProvider>
-      );
-    });
-    it('should pass mapped user to Provider', () => {
-      expect(wrapper).toHaveProp(
-        'value',
-        expect.objectContaining({
-          user: {
-            id: expect.any(String),
-            email: expect.any(String),
-            firstName: expect.any(String),
-            lastName: expect.any(String),
-            locale: expect.any(String),
-            timeZone: expect.any(String),
-            projects: expect.objectContaining({
-              total: 1,
-              results: expect.arrayContaining([
-                expect.objectContaining({ key: expect.any(String) }),
-              ]),
-            }),
-          },
-        })
-      );
-    });
-    it('should pass mapped project to Provider', () => {
-      expect(wrapper).toHaveProp(
-        'value',
-        expect.objectContaining({
-          project: {
-            key: expect.any(String),
-            version: expect.any(Number),
-            name: expect.any(String),
-            countries: expect.any(Array),
-            currencies: expect.any(Array),
-            languages: expect.any(Array),
-            ownerId: expect.any(String),
-          },
-        })
-      );
-    });
-    it('should pass mapped permissions to Provider', () => {
-      expect(wrapper).toHaveProp(
-        'value',
-        expect.objectContaining({
-          permissions: {
-            canManageProjectSettings: true,
-          },
-        })
-      );
+const renderAppWithContext = (ui: React.ReactElement) =>
+  render(
+    <ApplicationContextProvider<AdditionalEnvironmentProps>
+      user={createTestUser()}
+      project={createTestProject()}
+      environment={createTestEnvironment()}
+      projectDataLocale="en"
+    >
+      {ui}
+    </ApplicationContextProvider>
+  );
+
+describe('<ApplicationContext>', () => {
+  let rendered: RenderResult;
+  beforeEach(() => {
+    rendered = renderAppWithContext(
+      <ApplicationContext
+        render={context => (
+          <div>{`Project name: ${context.project &&
+            context.project.name}`}</div>
+        )}
+      />
+    );
+  });
+  it('should render project name from context', () =>
+    waitForElement(() => rendered.getByText('Project name: Foo 1')));
+});
+
+describe('useApplicationContext', () => {
+  let rendered: RenderResult;
+  const App = () => {
+    const projectName = useApplicationContext(
+      context => context.project && context.project.name
+    );
+    return <div>{`Project name: ${projectName}`}</div>;
+  };
+  beforeEach(() => {
+    rendered = renderAppWithContext(<App />);
+  });
+  it('should render project name from context', () =>
+    waitForElement(() => rendered.getByText('Project name: Foo 1')));
```

Comment:
Why do we need to waitFor here?

Reply:
Cause the context or project can be null on the first render

Reply:
To be clear: in the apps this is not a problem, cause the app gets rendered when the user/project are defined. However, since we're just testing the connector here, we need to account for this case.